src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvThread.cs

```diff
@@ -329,9 +330,7 @@ private void ThreadStart(object parameter)
             catch (Exception ex)
             {
                 _closeError = ex;
-                // Request shutdown so we can rethrow this exception
-                // in Stop which should be observable.
-                _appLifetime.StopApplication();
```

Comment:
We can't just remove this call to StopApplication() entirely. Looking at #12650, I see the idea was to remove this from the libuv transport library, but this logic still needs to exist somewhere in Kestrel. It seems that @davidfowl's idea was to have a thread error cause IConnectionListener.AcceptAsync to throw which would in turn cause Kestrel's ConnectionDispatcher to call StopApplication().

The StopApplication() call is here so the host knows to call IServer.Dispose() which in turn calls the LibuvThread.StopAsync() which rethrows the _closeError so the exception which originated on the libuv thread surfaces back to Program.Main().

Reply:
I'm happy to defer to your understanding. It was a little confusing because it seemed that the original comment said that we no longer needed to shutdown when the thread ends but then later at https://github.com/dotnet/aspnetcore/issues/12650#issuecomment-538856036 he mentions throwing the exception as in your understanding. 

I was worried about what would be the state if this happened in a primary listener with my understanding and implementation. I'll make your change now, it does seem safer.